### PR TITLE
Network Management System

### PR DESCRIPTION
**Pull request by john-breton**
Putting this up for review now so we can gather feedback ahead of time. Certain functionality needs to be refactored to interface with the ManagedRadioUnitRegistry and the RadioUnit creation classes.

### CODE REVIEW

File: src/main/java/mediator/MediatorIf.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package mediator;
+
+import java.beans.PropertyChangeListener;
+
+/**
+ * The Mediator interface serves as the communication gateway
+ * for all other components of the WirelessNetworkManagementSystem.
+ * <p>
+ * To view its implementation, please view {@link Mediator}
```
Comment (matthewkwsiu) [inline src/main/java/mediator/MediatorIf.java:9]: This is a cool tag that I didn't know about :)

File: src/main/java/mediator/MediatorIf.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * The Mediator interface serves as the communication gateway
+ * for all other components of the WirelessNetworkManagementSystem.
+ * <p>
+ * To view its implementation, please view {@link Mediator}
+ *
+ * @author ebreojh
+ */
+public interface MediatorIf {
+    void addPropertyChangeListener(PropertyChangeListener pcl);
+}
```
Comment (matthewkwsiu) [inline src/main/java/mediator/MediatorIf.java:16]: Very minor: There's a "no newline at end of file" warning here.

**Comment by john-breton** [comment]
If nobody objects, I'll merge this change within the next hour just so we have time to prepare the code for submission.

**Review by matthewkwsiu — COMMENTED** [review]
John, sorry for not having finished my review on Friday. I only got through a few files. Nothing really of note though. Here were some of my comments.

File: src/main/java/mediator/MediatorIf.java (hunk 2 above)
Comment (john-breton) [reply src/main/java/mediator/MediatorIf.java:16]: Oh shoot, thanks for catching that. I'll get that fixed for the next iteration.

**Pull request merged**